### PR TITLE
Feature/appointment

### LINKED ISSUE #10: 3.2.1 Calendar with booked appointments(4h)

### PR DESCRIPTION
**Pull request by VanHoanTran**
Closes #10

### CODE REVIEW

File: packages/mechanic-shop-fe/package.json [hunk 1]
```diff
@@ -11,9 +11,13 @@
     "@emotion/styled": "^11.10.5",
     "@mui/icons-material": "5.11.0",
     "@mui/material": "^5.11.5",
+    "date-fns": "^2.29.3",
+    "dayjs": "^1.11.7",
```
Comment (ghost) [inline packages/mechanic-shop-fe/package.json]: I don't think we need this, it's already a dependency of react-big-calendar, and I don't see it directly usedanywhere

File: packages/mechanic-shop-fe/package.json [hunk 2]
```diff
@@ -11,9 +11,13 @@
     "@emotion/styled": "^11.10.5",
     "@mui/icons-material": "5.11.0",
     "@mui/material": "^5.11.5",
+    "date-fns": "^2.29.3",
```
Comment (ghost) [inline packages/mechanic-shop-fe/package.json]: I don't see this used anywhere

File: packages/mechanic-shop-fe/src/components/AppointmentList/index.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { useSelector } from 'react-redux';
+import {
+  getAppointmentList,
+  getShopSettings,
+} from '../../reducers/queriesReducer';
+import paths from '../../paths';
+
+const localizer = momentLocalizer(moment);
+
+class Stall {
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/components/AppointmentList/index.js]: I don't think we need a class for this, as it is not stateful

File: packages/mechanic-shop-fe/src/components/AppointmentList/index.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  const shopSettings = useSelector(getShopSettings);
+
+  const stalls = useMemo(() => {
+    const { numberOfStalls } = shopSettings;
+    const results = [];
+    for (let i = 1; i <= numberOfStalls; i++) {
+      results.push(new Stall(i));
+    }
+    return results;
+  }, [shopSettings]);
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/components/AppointmentList/index.js]:
```suggestion
  const stalls = useMemo(() => new Array(shopSettings.numberOfStalls).fill().map((_, index) => ({
    stallId: index,
    stallTitle: `Stall ${index}`,
  })), [shopSettings]);
```

It is possible to create an empty array with a specific number of element, and then iterate through with map, transforming each of them to what we want, based on the index only

File: packages/mechanic-shop-fe/src/components/ShopLayout/SideBarInner/index.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import React, { useCallback, useMemo } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/components/ShopLayout/SideBarInner/index.js]: This change is not needed, useState is not used

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/mechanic-shop-fe/src/components/AppointmentList/index.js (hunk 4 above)
Comment (VanHoanTran) [reply packages/mechanic-shop-fe/src/components/AppointmentList/index.js]: this is really neat and clear. Thank you for suggestion

**Review by ghost — APPROVED** [review]
Nice work!

**Pull request merged**